### ISSUE TITLE
There is an issue with downloading events and then searching and storing coincidences

### ISSUE THREAD

**Issue by 153957** [HiSPARC/sapphire#76]
Using sapphire.download_data and sapphire.CoincidencesESD does not work.
It seems to be because the event_id with which the events are taken from their tables is a `numpy.uint64`, and apparently it is not allowed to get an index from a Table using a `numpy.uint64`.
The value should be an integer or numpy.uint32.
uint32 should be enough until you have more than 4,294,967,295 events, total number of events for 501 upto now: 203,679,321, so uint32 should be safe enough for now.
Im not yet certain where the uint64 came from.

**Comment by davidfokkema** [comment]
This is a recent problem, I guess?

**Comment by 153957** [comment]
Indeed, I did the same procedure (same script, different dataset) a couple of weeks ago without issue.
I guess I will do some git-bisecting.

**Comment by 153957** [comment]
Ah, its the same problem as we had before, this time the other way around.
Downgrading PyTables to 3.1 fixes the problem..

**Comment by 153957** [comment]
With PyTables 3.1:
```python
import tables
data = tables.open_file('muonlab_test3.h5', 'r')
data.root.station_501.events[11]
# (11L, 1433268013, 423254769L, 1433268013423254769L, [316, 417, 6, 5], [2824, 6445, 0, 0], 1.083899974822998, 2.317500114440918, 0.0, 0.0, 995.0, 1180.0, -999.0, -999.0, 1187.5)
data.root.station_501.events[numpy.uint64(11)]
# (11L, 1433268013, 423254769L, 1433268013423254769L, [316, 417, 6, 5], [2824, 6445, 0, 0], 1.083899974822998, 2.317500114440918, 0.0, 0.0, 995.0, 1180.0, -999.0, -999.0, 1187.5)
```

With PyTables 3.2:
```python
import tables
data = tables.open_file('muonlab_test3.h5', 'r')
data.root.station_501.events[11]
# (11L, 1433268013, 423254769L, 1433268013423254769L, [316, 417, 6, 5], [2824, 6445, 0, 0], 1.083899974822998, 2.317500114440918, 0.0, 0.0, 995.0, 1180.0, -999.0, -999.0, 1187.5)
data.root.station_501.events[numpy.uint64(11)]
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-64290e3841be> in <module>()
----> 1 data.root.station_501.events[numpy.uint64(11)]

tables/table.pyc in __getitem__(self, key)
   2111                 # To support negative values
   2112                 key += self.nrows
-> 2113             (start, stop, step) = self._process_range(key, key + 1, 1)
   2114             return self.read(start, stop, step)[0]
   2115         elif isinstance(key, slice):

tables/leaf.pyc in _process_range(self, start, stop, step, dim, warn_negstep)
    414         #if step is not None: step = long(step)
    415 
--> 416         return slice(start, stop, step).indices(long(nrows))
    417 
    418     _processRange = previous_api(_process_range)

TypeError: slice indices must be integers or None or have an __index__ method
```

**Comment by davidfokkema** [comment]
Buh

**Comment by 153957** [comment]
The uint64 array is created here: [sapphire/analysis/coincidences.py#L482](https://github.com/HiSPARC/sapphire/blob/5e986f5269ecbe81b519a43fa4e3b81e93999bd6/sapphire/analysis/coincidences.py#L482)
This is fixed in 6235d3639cae0c473c1cb29b9aa06f136e121a44 for CoincidencesESD.
The reason for uint64 is because the first element of each tuple in timestamps is the ext_timestamp.
However, this is not used when storing coincidences with CoincidencesESD.
Coincidences did store the timestamps array and therefore needed to convert it to an array that PyTables can handle, however the ext_timestamps in the timestamps array are no longer used after searching for coincidences.

I do see a small option for optimisation, currently each event in coincidences is read in _store_coincidence, only to read its ext_timestamp, timestamp and nanoseconds, while these are already available from the ext_timestamp in the timestamps array (except that those in the array may be changed by possible applied time shifts...).

**Comment by 153957** [comment]
Will be fixed on PyTables end in 3.2.2; PyTables/PyTables#496.

**Comment by 153957** [comment]
I can confirm this is fixed with PyTables 3.2.2.
The commits to work around this issue: a5662a493d4f82477beff87295d4c8b7a2686271 and part of 6235d3639cae0c473c1cb29b9aa06f136e121a44 could be reverted, but it should not make any significant difference.
